base/methodshow.jl

```diff
@@ -206,34 +205,49 @@ function sym_to_string(sym)
     end
 end
 
-function show(io::IO, m::Method)
+function show(io::IO, m::Method; modulecolor = :light_black, digit_align_width = -1)
     tv, decls, file, line = arg_decl_parts(m)
     sig = unwrap_unionall(m.sig)
     if sig === Tuple
         # Builtin
-        print(io, m.name, "(...) in ", m.module)
-        return
-    end
-    print(io, decls[1][2], "(")
-    join(
-        io,
-        String[isempty(d[2]) ? d[1] : string(d[1], "::", d[2]) for d in decls[2:end]],
-        ", ",
-        ", ",
-    )
-    kwargs = kwarg_decl(m)
-    if !isempty(kwargs)
-        print(io, "; ")
-        join(io, map(sym_to_string, kwargs), ", ", ", ")
-    end
-    print(io, ")")
-    show_method_params(io, tv)
-    print(io, " in ", m.module)
-    if line > 0
-        file, line = updated_methodloc(m)
-        print(io, " at ", file, ":", line)
+        print(io, m.name, "(...)")
+        file = "none"
+        line = 0
+    else
+        print(io, decls[1][2], "(")
+
+        # arguments
+        for (i,d) in enumerate(decls[2:end])
+            printstyled(io, d[1], color=:light_black)
+            if !isempty(d[2])
+                print(io, "::")
+                print_type_bicolor(io, d[2], color=:bold, inner_color=:normal)
+            end
+            i < length(decls)-1 && print(io, ", ")
+        end
+
+        kwargs = kwarg_decl(m)
+        if !isempty(kwargs)
+            print(io, "; ")
+            for kw in kwargs
+                skw = sym_to_string(kw)
+                if QuoteNode(kw) in m.roots # then it's required
```

Comment:
This is not a good way to test for this.

Reply:
IIUC, this just attempts to bold "required" keyword arguments? Since we don't really have a notion of required keyword arguments (just kwargs whose default is an error) I don't think this is essential and can just be left out.

Reply:
Jameson had the [exact same objection](https://github.com/JuliaLang/julia/pull/40251#discussion_r788208317) in the original PR by @mcabbott :sweat_smile:

Reply:
Yeah, that's what the code is supposed to do - and testing that explicitly makes it obvious that I didn't hit the last case `f(1,2,3; g=4)` with an extra keyword arg. Seems to be untested as well, so I'll add that test as well..

Reply:
Turns out I just forgot to be on the correct branch when testing that - the error is nicely formatted already.